networkx/algorithms/chordal.py

```
@@ -395,35 +407,110 @@ def complete_to_chordal_graph(G):
     >>> from networkx.algorithms.chordal import complete_to_chordal_graph
     >>> G = nx.wheel_graph(10)
     >>> H, alpha = complete_to_chordal_graph(G)
     """
+    if is_chordal(G):
+        H = G.copy()
+        alpha = dict.fromkeys(H, 0)
+    else:
+        H, alpha = _triangulate_fully(G)
+
+    return H, alpha
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+@nx._dispatchable
+def minimal_elimination_order(G):
+    """Return a copy of G triangulated to a minimal elimination ordering.
+
+    Does the same thing as `complete_to_chordal_graph`, but ensures that the
+    MCS-M algorithm is always run even if the graph is already chordal, in
+    contrast to the early return in `complete_to_chordal_graph`.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        Undirected graph
+
+    Returns
+    -------
+    H : NetworkX graph
+        The chordal enhancement of G
+    alpha : Dictionary
+            The elimination ordering of nodes of G
+    """
+    return _triangulate_fully(G)
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+@nx._dispatchable
+def perfect_elimination_order(G):
+    """Return a copy of G triangulated to a perfect elimination ordering.
+
+    A graph has a perfect elimination ordering if and only if it is chordal
+    (see the `is_chordal` documentation for more details); thus, an error is
+    raised if G is not chordal.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        Undirected graph
+
+    Returns
+    -------
+    H : NetworkX graph
+        The chordal enhancement of G
+    alpha : Dictionary
+            The elimination ordering of nodes of G
+
+    Raises
+    ------
+    NetworkXError
+        If the input graph is not chordal, a :exc:`NetworkXError` is raised.
+    """
+    if not is_chordal(G):
+        raise nx.NetworkXError("Input graph is not chordal.")
+
+    return _triangulate_fully(G)
+
+
+def _triangulate_fully(G):
+    # TODO: Add docstring
```

Is there any reason not to have `_triangulate_fully` be named `minimal_elimination_order` with its doc_string? That would perhaps make this slightly easier to read. But maybe I'm missing something.

---

+1 to me it seems like we could just have 1 method instead of 3

---

@dschult You're quite right; I'll just move the `_triangulate_fully` MCS-M logic into `minimal_elimination_order`. However, @amcandio, we still need all 3 of `complete_to_chordal_graph` (for backwards compatibility), `minimal_elimination_order` (for when we don't want `alpha` to be all zeros if the input is already chordal), and `perfect_elimination_order` (if we know our input is already chordal and want to raise an error if it is not).